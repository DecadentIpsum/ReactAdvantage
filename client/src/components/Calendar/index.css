.ui-chkbox-box.ui-state-active, .ui-radiobutton-box.ui-state-active {
  background: #263D5C !important; }

.ui-state-highlight {
  background: #5A88C3 !important; }

.ui-datepicker-trigger {
  background: #5A88C3 !important; }
