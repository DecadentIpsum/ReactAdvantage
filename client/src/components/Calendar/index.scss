@import "global-styles/common.scss";

.calendar {
    .ui-chkbox-box.ui-state-active,
    .ui-radiobutton-box.ui-state-active {
        background: $primaryDarker;
    }

    .ui-state-highlight {
        background: $primary;
    }

    .ui-datepicker-header {
        background: rgba($whiteShade, 0.6);
    }

    .ui-datepicker .ui-datepicker-prev {
        color: $blackShade;
    }

    .ui-datepicker .ui-datepicker-next {
        color: $blackShade;
    }

    .ui-inputtext {
      background: $white;
      color: $blackShade;
    }

    .ui-widget-content {
        background: $whiteShade;
        color: $blackShade;
    }

    .ui-datepicker.ui-widget .ui-datepicker-calendar td a {
        background: $white;
        color: $blackShade;
    }

    .ui-datepicker.ui-widget .ui-datepicker-calendar thead th {
        background: rgba($primary, 0.2);
        color: $blackShade;
    }
}