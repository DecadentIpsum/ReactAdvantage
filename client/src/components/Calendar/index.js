import React, { Component } from 'react';
import { Calendar as PrimeCalendar } from 'primereact/components/calendar/Calendar';

import './index.css';

export default class Calendar extends Component {
    render() {
        return (
            <div className="calendar">
                <PrimeCalendar
                    className="ui-datepicker-trigger"
                    showIcon
                    {...this.props}
                />
            </div>
        );
    }
}
