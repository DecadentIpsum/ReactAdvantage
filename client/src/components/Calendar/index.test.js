import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Calendar from './index';

it('renders correctly', () => {
    const component = shallow(<Calendar />);

    expect(component).toMatchSnapshot();
});
