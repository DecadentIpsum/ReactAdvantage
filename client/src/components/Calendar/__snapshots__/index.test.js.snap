// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calendar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Calendar
        appendTo={null}
        className="ui-datepicker-trigger"
        clearButtonClassName="ui-button-secondary"
        dataType="date"
        dateFormat="mm/dd/yy"
        defaultDate={null}
        disabled={false}
        disabledDates={null}
        disabledDays={null}
        hourFormat="24"
        icon="fa-calendar"
        id={null}
        inline={false}
        inputClassName={null}
        inputStyle={null}
        locale={
                Object {
                        "clear": "Clear",
                        "dayNames": Array [
                          "Sunday",
                          "Monday",
                          "Tuesday",
                          "Wednesday",
                          "Thursday",
                          "Friday",
                          "Saturday",
                        ],
                        "dayNamesMin": Array [
                          "Su",
                          "Mo",
                          "Tu",
                          "We",
                          "Th",
                          "Fr",
                          "Sa",
                        ],
                        "dayNamesShort": Array [
                          "Sun",
                          "Mon",
                          "Tue",
                          "Wed",
                          "Thu",
                          "Fri",
                          "Sat",
                        ],
                        "firstDayOfWeek": 0,
                        "monthNames": Array [
                          "January",
                          "February",
                          "March",
                          "April",
                          "May",
                          "June",
                          "July",
                          "August",
                          "September",
                          "October",
                          "November",
                          "December",
                        ],
                        "monthNamesShort": Array [
                          "Jan",
                          "Feb",
                          "Mar",
                          "Apr",
                          "May",
                          "Jun",
                          "Jul",
                          "Aug",
                          "Sep",
                          "Oct",
                          "Nov",
                          "Dec",
                        ],
                        "today": "Today",
                      }
        }
        maxDate={null}
        maxDateCount={null}
        minDate={null}
        monthNavigator={false}
        onBlur={null}
        onChange={null}
        onClearButtonClick={null}
        onContextMenu={null}
        onFocus={null}
        onKeyPress={null}
        onKeyUp={null}
        onMouseDown={null}
        onSelect={null}
        onTodayButtonClick={null}
        placeholder={null}
        readOnlyInput={false}
        required={false}
        selectOtherMonths={false}
        selectionMode="single"
        shortYearCutoff="+10"
        showButtonBar={false}
        showIcon={true}
        showOnFocus={true}
        showOtherMonths={true}
        showSeconds={false}
        showTime={false}
        stepHour={1}
        stepMinute={1}
        stepSecond={1}
        style={null}
        tabindex={null}
        timeOnly={false}
        todayButtonClassName="ui-button-secondary"
        utc={false}
        value={null}
        yearNavigator={false}
        yearRange={null}
/>,
      "className": "calendar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "appendTo": null,
        "className": "ui-datepicker-trigger",
        "clearButtonClassName": "ui-button-secondary",
        "dataType": "date",
        "dateFormat": "mm/dd/yy",
        "defaultDate": null,
        "disabled": false,
        "disabledDates": null,
        "disabledDays": null,
        "hourFormat": "24",
        "icon": "fa-calendar",
        "id": null,
        "inline": false,
        "inputClassName": null,
        "inputStyle": null,
        "locale": Object {
          "clear": "Clear",
          "dayNames": Array [
            "Sunday",
            "Monday",
            "Tuesday",
            "Wednesday",
            "Thursday",
            "Friday",
            "Saturday",
          ],
          "dayNamesMin": Array [
            "Su",
            "Mo",
            "Tu",
            "We",
            "Th",
            "Fr",
            "Sa",
          ],
          "dayNamesShort": Array [
            "Sun",
            "Mon",
            "Tue",
            "Wed",
            "Thu",
            "Fri",
            "Sat",
          ],
          "firstDayOfWeek": 0,
          "monthNames": Array [
            "January",
            "February",
            "March",
            "April",
            "May",
            "June",
            "July",
            "August",
            "September",
            "October",
            "November",
            "December",
          ],
          "monthNamesShort": Array [
            "Jan",
            "Feb",
            "Mar",
            "Apr",
            "May",
            "Jun",
            "Jul",
            "Aug",
            "Sep",
            "Oct",
            "Nov",
            "Dec",
          ],
          "today": "Today",
        },
        "maxDate": null,
        "maxDateCount": null,
        "minDate": null,
        "monthNavigator": false,
        "onBlur": null,
        "onChange": null,
        "onClearButtonClick": null,
        "onContextMenu": null,
        "onFocus": null,
        "onKeyPress": null,
        "onKeyUp": null,
        "onMouseDown": null,
        "onSelect": null,
        "onTodayButtonClick": null,
        "placeholder": null,
        "readOnlyInput": false,
        "required": false,
        "selectOtherMonths": false,
        "selectionMode": "single",
        "shortYearCutoff": "+10",
        "showButtonBar": false,
        "showIcon": true,
        "showOnFocus": true,
        "showOtherMonths": true,
        "showSeconds": false,
        "showTime": false,
        "stepHour": 1,
        "stepMinute": 1,
        "stepSecond": 1,
        "style": null,
        "tabindex": null,
        "timeOnly": false,
        "todayButtonClassName": "ui-button-secondary",
        "utc": false,
        "value": null,
        "yearNavigator": false,
        "yearRange": null,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Calendar
          appendTo={null}
          className="ui-datepicker-trigger"
          clearButtonClassName="ui-button-secondary"
          dataType="date"
          dateFormat="mm/dd/yy"
          defaultDate={null}
          disabled={false}
          disabledDates={null}
          disabledDays={null}
          hourFormat="24"
          icon="fa-calendar"
          id={null}
          inline={false}
          inputClassName={null}
          inputStyle={null}
          locale={
                    Object {
                              "clear": "Clear",
                              "dayNames": Array [
                                "Sunday",
                                "Monday",
                                "Tuesday",
                                "Wednesday",
                                "Thursday",
                                "Friday",
                                "Saturday",
                              ],
                              "dayNamesMin": Array [
                                "Su",
                                "Mo",
                                "Tu",
                                "We",
                                "Th",
                                "Fr",
                                "Sa",
                              ],
                              "dayNamesShort": Array [
                                "Sun",
                                "Mon",
                                "Tue",
                                "Wed",
                                "Thu",
                                "Fri",
                                "Sat",
                              ],
                              "firstDayOfWeek": 0,
                              "monthNames": Array [
                                "January",
                                "February",
                                "March",
                                "April",
                                "May",
                                "June",
                                "July",
                                "August",
                                "September",
                                "October",
                                "November",
                                "December",
                              ],
                              "monthNamesShort": Array [
                                "Jan",
                                "Feb",
                                "Mar",
                                "Apr",
                                "May",
                                "Jun",
                                "Jul",
                                "Aug",
                                "Sep",
                                "Oct",
                                "Nov",
                                "Dec",
                              ],
                              "today": "Today",
                            }
          }
          maxDate={null}
          maxDateCount={null}
          minDate={null}
          monthNavigator={false}
          onBlur={null}
          onChange={null}
          onClearButtonClick={null}
          onContextMenu={null}
          onFocus={null}
          onKeyPress={null}
          onKeyUp={null}
          onMouseDown={null}
          onSelect={null}
          onTodayButtonClick={null}
          placeholder={null}
          readOnlyInput={false}
          required={false}
          selectOtherMonths={false}
          selectionMode="single"
          shortYearCutoff="+10"
          showButtonBar={false}
          showIcon={true}
          showOnFocus={true}
          showOtherMonths={true}
          showSeconds={false}
          showTime={false}
          stepHour={1}
          stepMinute={1}
          stepSecond={1}
          style={null}
          tabindex={null}
          timeOnly={false}
          todayButtonClassName="ui-button-secondary"
          utc={false}
          value={null}
          yearNavigator={false}
          yearRange={null}
/>,
        "className": "calendar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "appendTo": null,
          "className": "ui-datepicker-trigger",
          "clearButtonClassName": "ui-button-secondary",
          "dataType": "date",
          "dateFormat": "mm/dd/yy",
          "defaultDate": null,
          "disabled": false,
          "disabledDates": null,
          "disabledDays": null,
          "hourFormat": "24",
          "icon": "fa-calendar",
          "id": null,
          "inline": false,
          "inputClassName": null,
          "inputStyle": null,
          "locale": Object {
            "clear": "Clear",
            "dayNames": Array [
              "Sunday",
              "Monday",
              "Tuesday",
              "Wednesday",
              "Thursday",
              "Friday",
              "Saturday",
            ],
            "dayNamesMin": Array [
              "Su",
              "Mo",
              "Tu",
              "We",
              "Th",
              "Fr",
              "Sa",
            ],
            "dayNamesShort": Array [
              "Sun",
              "Mon",
              "Tue",
              "Wed",
              "Thu",
              "Fri",
              "Sat",
            ],
            "firstDayOfWeek": 0,
            "monthNames": Array [
              "January",
              "February",
              "March",
              "April",
              "May",
              "June",
              "July",
              "August",
              "September",
              "October",
              "November",
              "December",
            ],
            "monthNamesShort": Array [
              "Jan",
              "Feb",
              "Mar",
              "Apr",
              "May",
              "Jun",
              "Jul",
              "Aug",
              "Sep",
              "Oct",
              "Nov",
              "Dec",
            ],
            "today": "Today",
          },
          "maxDate": null,
          "maxDateCount": null,
          "minDate": null,
          "monthNavigator": false,
          "onBlur": null,
          "onChange": null,
          "onClearButtonClick": null,
          "onContextMenu": null,
          "onFocus": null,
          "onKeyPress": null,
          "onKeyUp": null,
          "onMouseDown": null,
          "onSelect": null,
          "onTodayButtonClick": null,
          "placeholder": null,
          "readOnlyInput": false,
          "required": false,
          "selectOtherMonths": false,
          "selectionMode": "single",
          "shortYearCutoff": "+10",
          "showButtonBar": false,
          "showIcon": true,
          "showOnFocus": true,
          "showOtherMonths": true,
          "showSeconds": false,
          "showTime": false,
          "stepHour": 1,
          "stepMinute": 1,
          "stepSecond": 1,
          "style": null,
          "tabindex": null,
          "timeOnly": false,
          "todayButtonClassName": "ui-button-secondary",
          "utc": false,
          "value": null,
          "yearNavigator": false,
          "yearRange": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
